### PR TITLE
Helm chart linting and testing via helm/chart-tester action

### PR DESCRIPTION
**Pull request by smark88**
Added lint and test in integration-temp namespace. Simple go green test for now will work on more complex tests after the new year.

lint run https://github.com/broadinstitute/datarepo-helm/runs/1571725222
test run https://github.com/broadinstitute/datarepo-helm/pull/76/checks?check_run_id=1571725280

### CODE REVIEW

**Review by myessail — APPROVED** [review]

File: .github/test-config.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  - charts/sqlbackup
+  - charts/install-secrets-manager
+  - charts/cron-job
+  - charts/datarepomonitoring
+  - charts/gcp-managed-cert
+charts:
+  - charts/backend-config
+  - charts/create-secret-manager-secret
+  - charts/datarepo
+#  - charts/datarepo-api
```
Comment (nmalfroy) [inline .github/test-config.yaml:14]: Did you mean to leave these?

File: charts/create-secret-manager-secret/values.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
 secrets:
-- secretName:      ## name of kubeSecret
-  nameSpace:
-  vals:
-  - kubeSecretKey:      ## name of kubeSecretKey
-    path:     ## vault path
-    encoding:     ## if secret is base64 already use this flagg
-    vaultKey:     ## vault key name
+
+# - secretName:     ## name of kubeSecret
```
Comment (nmalfroy) [inline charts/create-secret-manager-secret/values.yaml:3]: Probably should just delete

File: charts/datarepo-api/ci/testing-values.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+---
+# overrides an application.properties item in DRmanager
+image:
+  tag: 1.0.180
```
Comment (nmalfroy) [inline charts/datarepo-api/ci/testing-values.yaml:4]: How is this tag value going to be kept in sync (same Q for the UI chart)

File: charts/oidc-proxy/ci/testing-values.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+fullnameOverride: test-oidc-proxy
+env:
+  PROXY_URL: http://test-datarepo-ui.integration-temp:8080/
+  PROXY_URL2: http://test-datarepo-api.integration-temp:8080/api
+  PROXY_URL3: http://test-datarepo-api.integration-temp:8080/register
+  LOG_LEVEL: debug
+  SERVER_NAME: jade.datarepo-integration.broadinstitute.org
+  REMOTE_USER_CLAIM: sub
+  ENABLE_STACKDRIVER: yes
```
Comment (nmalfroy) [inline charts/oidc-proxy/ci/testing-values.yaml:9]: huh interesting..very meta...did the lintintg PR catch a linting issue with itself?

**Review by nmalfroy — APPROVED** [review]
Looks good.  I have a couple of questions on how stuff gets updated and there's some commented code

File: .github/test-config.yaml (hunk 1 above)
Comment (smark88) [reply .github/test-config.yaml:14]: I will move them to excluded as the individual charts cannot be tested by itself but the umbrella can be selected to be tested.

File: charts/datarepo-api/ci/testing-values.yaml (hunk 3 above)
Comment (smark88) [reply charts/datarepo-api/ci/testing-values.yaml:4]: might be worth making an action for updating the chart default so wee dont need to set it.

File: charts/oidc-proxy/ci/testing-values.yaml (hunk 4 above)
Comment (smark88) [reply charts/oidc-proxy/ci/testing-values.yaml:9]: not my docker image but it wants yes lol

File: .github/postgres-values.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+rbac:
+  create: true
+persistence:
+  enabled: true
+initdbUser: postgres
+initdbPassword: postgres
+initdbScripts:
+  init.sql: |
+    CREATE USER drmanager WITH PASSWORD 'postgres';
+    CREATE DATABASE cheese WITH OWNER drmanager;
```
Comment (fboulnois) [inline .github/postgres-values.yaml:21]: Do we also need the `cheese` database?

**Review by fboulnois — COMMENTED** [review]
Overall lgtm. Should we delete the `.bak` file? One additional question:

File: .github/postgres-values.yaml (hunk 5 above)
Comment (smark88) [reply .github/postgres-values.yaml:21]: nope lol

**Review by fboulnois — APPROVED** [review]
lgtm

**Comment by smark88** [comment]
Final edit please review

**Review by fboulnois — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        run: |
+          gcloud container clusters get-credentials ${{ env.K8_CLUSTER }} --zone ${{ env.GOOGLE_ZONE }}
+          gcloud info
+
+      - name: supporting postgres chart
+        timeout-minutes: 3
+        run: |
+          helm upgrade test bitnami/postgresql -f .github/postgres-values.yaml --install -n ${{ env.NAMESPACE }} --create-namespace
+          while [[ $(kubectl get pods -l app.kubernetes.io/name=postgresql -n ${{ env.NAMESPACE }} -o \
+            'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do echo "waiting for pod" && sleep 1; done
```
Comment (nmalfroy) [inline .github/workflows/test.yaml:92]: Cool, so this is using the pod's readiness state?

**Review by nmalfroy — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 6 above)
Comment (smark88) [reply .github/workflows/test.yaml:92]: yeah mang

**Pull request merged**